doc/chapters/configuration/index.rst

```diff
@@ -49,4 +49,33 @@ As it is indicated in :ref:`design chapter <design>`, each IPC is managed by a g
 
 CMake Options
 =============
-CMake options can be used to configure softwareContainer. Desired gateways can be enabled/disabled by passing flags to cmake. (Currently following gateways are able for enable/disable option; network, pulse, device node, dbus, cgroup :ref:`Gateways <gateways>`).
+
+CMake options can be used to configure softwareContainer. The CMake options can be categorized into four groups:
+
+1. Enabling / disabling gateways.
+2. Enabling / disabling generation of documentation.
+3. Network
+4. Testing
+
+Gateways options
+----------------
+If a gateway is disabled this means that the gateway will not be apart of SoftwareContainer at compile time. Hence this will disable the IPC mechanism that the gateway in question handles.
+As an example, the following snippet will install SoftwareContainer without any network support::
+
+    cmake -DENABLE_NETWORKGATEWAY=OFF ..
+    make
+    sudo make install
+
+A complete list of available gateways can be found at :ref:`Gateways <gateways>`.
+
```

Comment:
We discussed this a bit yesterday. The risk of getting slightly skewed docs increase the more duplication we have. Maybe it can be OK with referencing the root README.md and some other part of the user-docs that shows an examples to allow the reader to put the pieces together. E.g. reference the README early, describe what sort of things are configured with CMake and then also reference a more complete example. Maybe it's possible to do in a way that the reader can put the two pieces together and figure out how to specify the different options. 

Or we go the other way and duplicate in such a way that it's easily grep'able.

My basic idea of this is essentially that the user should reference the root README.md for configuring and building SC itself, and the user-docs will explain other configuration options, e.g. the configuration files, and how it all ties together. One is totally configure-and-build centric, the other use-case centric.